File: Python/P07.py
#Problem 7
#By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
#What is the 10 001st prime number?
#Answer:104743

def calc(n):
    if n == 2 or n == 3: return True
    if n % 2 == 0 or n < 2: return False
    for i in range(3, int(n**0.5) + 1, 2):
        if n % i == 0:
            return False
    return True

t,n = 0,2
while t != 10001:
    t += calc(n)
    n += 1

if __name__ == "__main__":
    print(n - 1)

File: Java/P07.java
/*
Problem 7
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
What is the 10 001st prime number?
Answer:104743
*/

public class P07 {
	public static void main(String[] args) 
	{
		int p=2, s=0, c;
		for(int i=3; p<=10001;i+=2)
			{
			c=0;
				for(int j=3;j<=Math.sqrt(i);j+=2)
					{
					if(i%j==0 && i!=j)
					{
						c++;
					}
					}
				if(c==0)
				{
					s=i;
					++p;
				}
				}
		System.out.println(s);}
}
